File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();
        if(n==0) return 0;
        unordered_set<int> st;
        for(auto ele:nums) st.insert(ele);
        int longest=1;
        for(auto ele:st){
            if(st.find(ele-1)==st.end()){
                int cnt=1;
                int x=ele;
                while(st.find(x+1)!=st.end()){
                    x=x+1;
                    cnt++;
                }
                longest=max(longest,cnt);
            }
        }
        return longest;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        if len(nums)==0:
            return 0
        seen=set()
        for ele in nums:
            seen.add(ele)
        max_ans=float('-inf')
        for ele in seen:
            if (ele-1) in seen:
                continue
            else:
                cnt=0
                while ele in seen:
                    cnt+=1
                    ele+=1
                max_ans=max(max_ans,cnt)

        return max_ans
